README.md

```diff
@@ -5,79 +5,116 @@
+To get a logger, one can use the following:
 
-If you prefer a backend other than java.util.logging, add dependencies that include a `runtimeOnly`
-dependency for your chosen logging backend. For instance:
+1. Make a logging class implement `WithLogging` interface.
+2. Get a logger from `LoggingFactory`.
+
+The interface provides a default method `logger()` that returns a logger
+for the implementing class or object:
 
 ```kotlin
-dependencies {
-    implementation("io.spine:spine-logging:$version")
-    runtimeOnly("com.google.flogger:flogger-log4j2-backend:$floggerVersion")
+import io.spine.logging.WithLogging
+
+class Example : WithLogging {
+    fun doSomething() {
+        logger() // Call to the default method of `WithLogging`.
+            .atWarning()
+            .log { "..." }
+    }
 }
 ```
 
-For SLF4J as a backend, your dependencies should also include a backend for SLF4J. For example:
+`LoggingFactory` has two methods that return a logger for the enclosing class
+and for the given `KClass`:
+
+```kotlin
+import io.spine.logging.LoggingFactory
+
+class App {
+    private val logger1 = LoggingFactory.forEnclosingClass()
+    private val logger2 = LoggingFactory.loggerFor(this::class)
+
+    fun doSomething() {
+        check(logger1 === logger2) // There is always one logger per class.
+        logger1.atWarning()
+            .log { "..." }
+    }
+}
+```
+
+## Logging backends
+
+A logging backend handles an actual output of the logged statement. 
+`LogRecord` may be printed to the console, be written to a file or sent by 
+the network to some log-aggregating service. It all is up to a chosen backend 
+and its configuration.
+
+The following backends are available:
+
+* `io.spine:spine-logging-backend` – the default JUL-based backend.
+* `io.spine:spine-logging-log4j2-backend` – Log4j2 backend.
+
+Put a chosen backend to `runtimeOnly` configuration, and the logging library
+will discover it in the runtime.
+
+An example usage of Log4j2 backend:
 
 ```kotlin
 dependencies {
     implementation("io.spine:spine-logging:$version")
-    runtimeOnly("com.google.flogger:flogger-slf4j-backend:$floggerVersion")
-    runtimeOnly("org.slf4j:slf4j-reload4j:$slf4jVersion")
+    runtimeOnly("io.spine:spine-logging-log4j2-backend:$version")
 }
 ```
 
-## Logging context
+Please note, only one backend implementation should be present in the runtime.
+Two or more backends will cause an exception because the logging framework 
+will not be able to understand, which one should be used.
+
+## Logging contexts
 
-A logging context refers to a set of attributes that are attached to all log records while
-a context is installed. For instance, you can attach a user ID to all log records for
-the current request.
+A logging context refers to a set of attributes that are attached to all log 
+records while a context is installed. For instance, you can attach a user ID 
+to all log records for the current request.
```

Comment:
The last sentence probably deserves a code snippet. But I don't insist at the moment.

Reply:
So, I've decided to add a snippet because it's unclear how to use custom contexts. There's no example anywhere.

BTW, I've not tried them before.

It works, but there are several suggestions: #70, #71, #72.